Add tutorials for blend animation and retargeting (#6322)
godotengine/godot-docs: tutorials/animation/animation_tree.rst

* *Priority* is used together with the ``travel()`` function from code (more on this later). Lower priority transitions are preferred when travelling through the tree.
* *Disabled* toggles disabling this transition (when disabled, it will not be used during travel or auto advance).

For better blending
-------------------

> In Godot 4.0+, for blending results consistency, the property values to be blended always have a specific initial value. For example, in the case of two animations to be blended, if one animation has a property track and the other does not, the blended animation is calculated as if the latter animation had a property track with the initial value.

Reviewer: for blending results consistency, -> for blending results consistently,

> In Godot 4.0+, for blending results consistency, the property values to be blended always have a specific initial value. For example, in the case of two animations to be blended, if one animation has a property track and the other does not, the blended animation is calculated as if the latter animation had a property track with the initial value.

Reviewer: ```suggestion
In Godot 4.0+, for blending results consistently, the property values to be blended always have a specific initial value. For example, in the case of two animations to be blended, if one animation has a property track and the other does not, the blended animation is calculated as if the latter animation had a property track with the initial value.
```

> For Position/Rotation/Scale 3D tracks for Skeleton3D bones, the initial value is Bone Rest. For other properties, the initial value is ``0`` (exactly the value of ``zero()`` as Variant) and if present the track value in RESET animation, it's first key value is used instead.

Reviewer: ```suggestion
When using Position/Rotation/Scale 3D tracks for Skeleton3D bones, the initial value is Bone Rest. For other properties, the initial value is ``0`` (exactly the value of ``zero()`` as Variant) and if present the track value in RESET animation, it's first key value is used instead.
```

.. image:: img/blending1.png

This means that the animation lacking that will treat those Positions as ``Vector2(0, 0)``.

.. image:: img/blending2.gif

This problem can be solved by adding a Property track for Position as an initial value to the RESET animation.

.. image:: img/blending3.png

.. image:: img/blending4.gif

.. note:: Be aware that the RESET animation exists to define the default pose when loading an object originally. It is assumed to have only one frame and is not expected to be played back using the timeline.

Also keep in mind that the Rotation 3D tracks and the Property tracks for 2D rotation with Interpolation Type set to Linear Angle or Cubic Angle will prevent rotation more than 180 degrees from the initial value as blended animation.

This behavior is useful for Skeleton3D to prevent the bones penetrating the body when blend animations. Therefore, Skeleton3D's Bone Rest values should be as close to the midpoint of the movable range as possible. **This means that for humanoid models, it is preferable to import them with T-pose**.

.. image:: img/blending5.gif

> You can see that the shortest rotation path from Bone Rests is taken priority rather than the shortest rotation path between animations.

Reviewer: ```suggestion
You can see that the shortest rotation path from Bone Rests is prioritized rather than the shortest rotation path between animations.
```

* *Priority* is used together with the ``travel()`` function from code (more on this later). Lower priority transitions are preferred when travelling through the tree.
* *Disabled* toggles disabling this transition (when disabled, it will not be used during travel or auto advance).

For better blending
-------------------

> In Godot 4.0+, for blending results consistently, the property values to be blended always have a specific initial value.

Reviewer 2: ```suggestion
In Godot 4.0+, for blending results consistently, the property values to be blended always have a specific initial value.
```
Grammar feels funky here, but I'm not enough of a natural speaker to provide a suggestion.
Maybe "for blending results to be consistent", or maybe restructure the whole sentence. Not sure. Hope someone else has a suggestion :)

Author: @mhilbrunner In Godot 3, when blending was performed during animation playback, the blending result depended on when the blending was performed.

I want to mean that the results of blending always be equal.

> In Godot 4.0+, for blending results consistently, the property values to be blended always have a specific initial value.

Author: I don't think it is necessary to mentions even the internal process (the setting order and the connecting order) since it increases the complexity.

> In Godot 4.0+, for blending results consistently, the property values to be blended always have a specific initial value.

Author: How about this?

> In Godot 4.0+ to make blending results reproducible and consistent always, the blended property values to be blended must have a specific initial value.

> In Godot 4.0+, for blending results consistently, the property values to be blended always have a specific initial value.

Reviewer: Try

The blended property values must have a specific initial value to make blending results reproducible and consistent in Godot.

> In Godot 4.0+, for blending results consistently, the property values to be blended always have a specific initial value.

Reviewer 3: I may try building this tutorial within a few days, and trying it out to give feedback. Until then, I cannot resist not giving some feedback on that phrase

>In Godot 4.x, in order for the blending results to be deterministic (reproducible and always consistent), the blended property values must have a specific initial value

For example, in the case of two animations to be blended, if one animation has a property track and the other does not,
the blended animation is calculated as if the latter animation had a property track with the initial value.

When using Position/Rotation/Scale 3D tracks for Skeleton3D bones, the initial value is Bone Rest.

> For other properties, the initial value is ``0`` (exactly the value of ``zero()`` as Variant)

Reviewer 2: ```suggestion
(exactly the value of ``zero()`` as Variant)
```
Is this a neccessary detail? It IMO makes an already complicated topic/sentence more complex.

Author: @mhilbrunner Would you recommend that I remove the `zero()` statement?

It was meant to suggest that there are two initial values for Variant, init() and zero(), but that we need to use zero() for correct calculation, but it is okay to remove it.

Position/Rotation/Scale 3D tracks for bones use Bone Rest. In this case, Bone Rest is implicitly and fundamentally never set to 0 on import.

However, Property tracks for changing other properties may become 0 unless RESET animation is explicitly set.

Author: Removed mention about zero().

and if the track is present in the ``RESET`` animation, the value of its first keyframe is used instead.

For example, the following AnimationPlayer has two animations, but one of them lacks a Property track for Position.

.. image:: img/blending1.png

This means that the animation lacking that will treat those Positions as ``Vector2(0, 0)``.

.. image:: img/blending2.gif

This problem can be solved by adding a Property track for Position as an initial value to the ``RESET`` animation.

.. image:: img/blending3.png

.. image:: img/blending4.gif

.. note:: Be aware that the ``RESET`` animation exists to define the default pose when loading an object originally.
          It is assumed to have only one frame and is not expected to be played back using the timeline.

Also keep in mind that the Rotation 3D tracks and the Property tracks for 2D rotation

> with Interpolation Type set to Linear Angle or Cubic Angle will prevent rotation more than 180 degrees

Reviewer 2: ```suggestion
with Interpolation Type set to Linear Angle or Cubic Angle will prevent rotation of more than 180 degrees
```

from the initial value as blended animation.

> This behavior is useful for Skeleton3D to prevent the bones penetrating the body when blend animations.

Reviewer 2: ```suggestion
This can be useful for Skeleton3Ds to prevent the bones penetrating the body when blending animations.
```

Therefore, Skeleton3D's Bone Rest values should be as close to the midpoint of the movable range as possible.

> **This means that for humanoid models, it is preferable to import them with T-pose**.

Reviewer 2: ```suggestion
**This means that for humanoid models, it is preferable to import them in a T-pose**.
```

> .. image:: img/blending5.gif

Reviewer 2: Before merging this, the docs team needs to make a decisions on the move to .webp, as an animated webp could potentially save a lot of size here compared to a 4-5mb gif.

Author: I pushed webp to test it out.
